File: 0189-rotate-array/0189-rotate-array.py
class Solution:
    def rotate(self, nums: List[int], k: int) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """

        def reverseArr(i , j):
            while i < j:
                nums[i] , nums[j] = nums[j] , nums[i]
                i += 1
                j -= 1
            
        n = len(nums)
        rot = k % n
        if rot == 0: return 

        reverseArr(0 , n- 1)
        reverseArr(rot , n - 1)
        reverseArr(0 , rot - 1)

       
        
        

    
        

File: 0189-rotate-array/0189-rotate-array.go
func rotate(nums []int, k int)  {
    
    reverseArr := func(arr []int, i , j int ){
        for i < j{
            arr[i], arr[j] = arr[j], arr[i]
            i += 1
            j -= 1
        }
    }
    n := len(nums)
    rot := k % n
    if rot == 0 {
        return 
    }
    
    reverseArr(nums, 0, n - 1);
    reverseArr(nums, rot , n - 1);
    reverseArr(nums, 0 , rot-1);

    
}